framework/src/utils/RankTwoTensor.C

```diff
@@ -889,6 +889,256 @@ RankTwoTensorTempl<T>::symmetricEigenvaluesEigenvectors(std::vector<T> & eigvals
       eigvecs(j, i) = a[i * N + j];
 }
 
+template <typename T>
+void
+RankTwoTensorTempl<T>::adSymmetricEigenvaluesEigenvectors(RankTwoTensorTempl<T> & Q,
```

Comment:
Don't add a separate method. Instead make this a template specialization for `DualReal`

```
template <>
void
RankTwoTensorTempl<DualReal>::symmetricEigenvaluesEigenvectors(std::vector<DualReal> & eigvals, RankTwoTensorTempl<DualReal> & eigvecs) const
{
...
}
```